NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_refresh.cpp

```diff
@@ -50,6 +50,119 @@ namespace BE = BiometricEvaluation;
 
 std::mutex mutGray;
 
+// Wrappers for yesOrNo Prompts
+bool
+NFIQ2UI::askIfQuantize()
+{
+	const std::string prompt =
+	    "This Image is not 8 bit or 1 bit grayscale. Would you like to "
+	    "quantize/convert this image to 8 bit color and depth?";
+	return NFIQ2UI::yesOrNo(prompt, false, true, true);
+}
+
+bool
+NFIQ2UI::askIfDefaultResolution(const std::string &name,
+    const uint16_t defaultDPI, const uint16_t requiredDPI)
+{
+	// 72PPI is default resolution
+	if (defaultDPI != 72) {
+		return false;
+	}
+	const std::string prompt { "The resolution of \"" + name +
+		"\" was parsed as " + std::to_string(defaultDPI) +
+		" PPI, which is "
+		"sometimes used to indicate that resolution information was not "
+		"recorded. NFIQ 2 only supports " +
+		std::to_string(requiredDPI) +
+		" PPI images. This application can resample images, but doing so "
+		"introduces error.\n"
+		"Assume this image was actually captured at " +
+		std::to_string(requiredDPI) + " PPI?" };
+	return NFIQ2UI::yesOrNo(prompt, false, true, true);
+}
+
+bool
+NFIQ2UI::askIfResample(const std::string &name, const uint16_t imageDPI,
+    const uint16_t requiredDPI)
+{
+	const std::string prompt { "The resolution of \"" + name +
+		"\" was parsed as " + std::to_string(imageDPI) + " PPI, not " +
+		std::to_string(requiredDPI) +
+		" PPI, as required for NFIQ 2. "
+		"Would you like to introduce error and re-sample this "
+		"image to " +
+		std::to_string(requiredDPI) + " PPI?" };
+	return NFIQ2UI::yesOrNo(prompt, false, true, true);
+}
+
+cv::Mat
+NFIQ2UI::resampleAndLogError(BE::Memory::uint8Array &grayscaleRawData,
+    const NFIQ2UI::DimensionInfo &dimensionInfo,
+    const NFIQ2UI::ImageProps &imageProps,
+    std::shared_ptr<NFIQ2UI::Log> logger = nullptr)
```

Comment:
Did this compile? Default params go in the header.

```suggestion
    std::shared_ptr<NFIQ2UI::Log> logger)
```